GameData/RP-1/Contracts/X-Plane Research/XPlanesSupersonicOptionalHigh.cfg

```diff
@@ -72,22 +74,51 @@ CONTRACT_TYPE
 		type = CompleteContract
 		contractType = XPlanesSupersonicMach2
 	}
-
+	
+	REQUIREMENT
+	{
+		name = AcceptContract
+		type = AcceptContract
+		tag = exclude_Supersonic
+		invertRequirement = true
+	}
+	
 	BEHAVIOUR
 	{
-		name = IncrementTheCount
+		name = IncrementTheCount 
 		type = Expression
 
 		CONTRACT_OFFERED
 		{
+			
 			XPSS_Completion = ($XPSS_Completion + 0) == 0 ? (UniversalTime() - @RP0:expectedDays_XPlanesSupersonicOptional * 86400) : ($XPSS_Completion + 0)
+		
 		}
+		
 		CONTRACT_COMPLETED_SUCCESS
 		{
-			RP0_XPSS_Difficulty = $RP0_XPSS_Difficulty + 1
 			XPSS_Completion = UniversalTime()
 		}
 	}
+	
+	BEHAVIOUR
+	{
+		name = SetSSDifficulty
+		type = Expression
+
+
+		CONTRACT_COMPLETED_SUCCESS
+		{
+			RP0_XPSS_High_Difficulty = $RP0_XPSS_High_Difficulty + 1
+		}
+	}
+	
+	DATA
+	{
+		type = List<float>
+		minSpeedKMH = [ 650, 700, 750, 800, 850, 900, 950, 1000, 1050, 1100, 1150, 1200]
```

Comment:
I really do not believe km/h is the correct unit here. 🤔

Reply:
You're right, I'll fix that.